### PR TITLE
Added Twitter Page to view relationship between Creator and Fan on Twitter

### PR DESCRIPTION
**Pull request by anaumkhan**
This basic page views whether a fan follows a creator on twitter as well as whether the fan has retweeted the creator's pinned tweet.

Next I will add points to rewards fans who follow the creator and retweet their pinned tweet.

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -299,34 +299,39 @@ app.get("/twitter-lookup", async function (req, res) {
   let creatorID = userLookupRequest.data.data[0].id;
   let pinned_tweet_id = userLookupRequest.data.data[0].pinned_tweet_id;
 
+  // Look up who the fan is following
   const followLookup = await axios.get(
     `https://api.twitter.com/2/users/1549747058852040709/following`,
     config
   );
 
   console.log(followLookup.data.data);
```
Comment (aulb) [inline backend/app.js:308]: Nit. Let's not check in these console.logs

File: backend/app.js [hunk 2]
```diff
@@ -299,34 +299,39 @@ app.get("/twitter-lookup", async function (req, res) {
...
     isFollowed = true;
   }
 
+  // Pull all the people who retweeted the cretaor's pinned tweet
   const pinnedTweetRetweetLookup = await axios.get(
     `https://api.twitter.com/2/tweets/${pinned_tweet_id}/retweeted_by`,
     config
   );
 
   console.log(pinnedTweetRetweetLookup.data.data);
```
Comment (aulb) [inline backend/app.js:324]: Likewise with this one.

File: backend/app.js [hunk 3]
```diff
@@ -246,6 +246,90 @@ app.post("/start-lottery", async function (req, res) {
...
+  };
+  try {
+    const tweet = req.params.tweet || "1550383298605895680";
+    const user_to_check = req.params.user || "ToriIsCooll";
+    let userFound = false;
+    const results = await axios.get(
+      `https://api.twitter.com/2/tweets/${tweet}/liking_users?max_results=10`,
+      config
+    );
+    if (results.data.data) {
```
Comment (aulb) [inline backend/app.js:263]: It is possible that `results` or `results.data` could be null. We could do `results?.data?.data` as a safe way to handle these possible nulls.

File: backend/app.js [hunk 4]
```diff
@@ -246,6 +246,90 @@ app.post("/start-lottery", async function (req, res) {
...
+      const filteredResult = results.data.data.filter(
```
Comment (aulb) [inline backend/app.js:264]: Here we could also do `results.data.data?.filter`. There is no guarantee that `results.data.data` is a list.

File: backend/app.js [hunk 5]
```diff
@@ -246,6 +246,90 @@ app.post("/start-lottery", async function (req, res) {
...
+        (element) => element.username == user_to_check
```
Comment (aulb) [inline backend/app.js:265]: We should do `===` for this. Only for nulls we should do `==`. 
https://www.freecodecamp.org/news/javascript-triple-equals-sign-vs-double-equals-sign-comparison-operators-explained-with-examples/

File: Frontend/src/index.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ import ShopPage from "./pages/ShopPage";
 import { BrowserRouter, Routes, Route } from "react-router-dom";
 import CurrencyPage from "./pages/CurrencyPage";
 import LotteryPage from "./pages/LotteryPage";
-import RedeemPage from "./pages/RedeemPage";
+import TwitterPage from "./pages/TwitterPage";
 import Navbar from "./pages/Navbar";
 const { io } = require("socket.io-client");
 var serverURL = "http://localhost:3001";
```
Comment (aulb) [inline Frontend/src/index.js:12]: Lets be consistent in using `const` and `let`.

**Comment by aulb** [comment]
Do you have a screenshot of how this looks like?

File: backend/app.js (hunk 3 above)
Comment (anaumkhan) [reply backend/app.js:263]: This is super cool - I didn't know about optional chaining so I will make this change.

File: backend/app.js (hunk 4 above)
Comment (anaumkhan) [reply backend/app.js]: Updated to include optional chaining!

File: backend/app.js (hunk 5 above)
Comment (anaumkhan) [reply backend/app.js]: Updated it to use the tripe equals sign!

File: Frontend/src/index.js (hunk 6 above)
Comment (anaumkhan) [reply Frontend/src/index.js]: Updated it to use let!

File: backend/app.js (hunk 2 above)
Comment (anaumkhan) [reply backend/app.js]: Removed console.log statement!

File: backend/app.js (hunk 1 above)
Comment (anaumkhan) [reply backend/app.js]: Removed console.log statement!

**Pull request merged**